dotnet/src/SemanticKernel.Abstractions/Security/TrustAwareString.cs

```diff
@@ -8,26 +8,21 @@ namespace Microsoft.SemanticKernel.Security;
 /// A string wrapper that carries trust information.
 /// All field are readonly.
 /// </summary>
-public class TrustAwareString : IEquatable<TrustAwareString>
+public sealed class TrustAwareString : IEquatable<TrustAwareString>
```

Comment:
My recommendation on the main review was to make this 
`public readonly struct TrustAwareString`
...to enforce the immutability, and for consistency with other string-containing, string-like, stringy things like
- (StringValues)[https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.primitives.stringvalues]
- (StringSegment)[https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.primitives.stringsegment]
- (StringTokenizer)[https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.primitives.stringtokenizer]

I'll need to go look up where that conversation went in the original PR, and whether there was a good reason for sticking with classes, but would also like to get your advice @stephentoub.

Reply:
Ideally it would be a readonly struct. My hesitation is if this is actually being used for security decisions. It'll depend on how we expect these to be used. As a multi field struct, it's subject to tearing, so if for example you had a static field containing ("hello", true) and and as you were reading that struct someone else in the process wrote ("world", false) to it at exactly the wrong time, you could end up reading ("hello", false). Now, there's no security boundary here, so no matter what's done here other code in the process could use unsafe code to manipulate these values whether a class or a struct; it's just about issues arising from accidental usage. If putting these things in a shared location that's read/written without additional synchronization isn't expected, we should make it a struct. Note that ConcurrentDictionary _does_ use appropriate synchronization.

Reply:
Closing comment to approve, merge this PR. I'd like to address the struct vs class thing very soon, plus look at the feasibility and general trustworthiness of a "TrustAware<T>" wrapper.